implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/types/DynatraceLongTaskTimerTest.java

```diff
@@ -0,0 +1,153 @@
+            .isCloseTo(snapshot.getMax(), TOLERANCE);
+    }
+
+    /**
+     * This test *could* be done with the MockClock, but it gets pretty unintuitive. That
+     * is because when adding to the MockClock in one Thread, it automatically adds to the
+     * other thread as well and time is basically "double-counted".
+     */
+    @Test
+    void testTwoValuesAreRecorded() throws InterruptedException {
+        // use the system clock, as it is much easier to understand than when using the
+        // MockClock (When using MockClock, adding to the clock in two separate threads
+        // basically means that two things happen at the same time).
+        DynatraceLongTaskTimer ltt = new DynatraceLongTaskTimer(ID, Clock.SYSTEM, TimeUnit.MILLISECONDS,
+                DISTRIBUTION_STATISTIC_CONFIG, false);
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        // both tasks need to be running for a while before we take the snapshot
+        CountDownLatch taskHasBeenRunningLatch = new CountDownLatch(2);
+        CountDownLatch stopLatch = new CountDownLatch(1);
+
+        executorService.submit(() -> {
+            ltt.record(() -> {
+                try {
+                    Thread.sleep(70);
+                    taskHasBeenRunningLatch.countDown();
+
+                    // wait until the snapshot has been taken.
+                    stopLatch.await(300, TimeUnit.MILLISECONDS);
+                }
+                catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        });
+
+        executorService.submit(() -> {
+            ltt.record(() -> {
+                try {
+                    Thread.sleep(30);
+                    taskHasBeenRunningLatch.countDown();
+
+                    // wait until the snapshot has been taken.
+                    stopLatch.await(300, TimeUnit.MILLISECONDS);
+                }
+                catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        });
+
+        taskHasBeenRunningLatch.await(300, TimeUnit.MILLISECONDS);
+
+        DynatraceSummarySnapshot snapshot = ltt.takeSummarySnapshot();
+        // can release the background tasks
+        stopLatch.countDown();
+
+        // the first Thread has been "running" for ~30ms at the time of recording and will
+        // supply the min
+        assertThat(snapshot.getMin()).isGreaterThanOrEqualTo(30);
+        // the second Thread has been "running" for ~70ms at the time of recording and
+        // will supply the max
+        assertThat(snapshot.getMax()).isGreaterThanOrEqualTo(70).isLessThan(100);
+        // the min is greater than 30, and the max is greater than 70, so together they
+        // have to be greater than 100.
+        assertThat(snapshot.getTotal()).isGreaterThan(100);
+        assertThat(snapshot.getCount()).isEqualTo(2);
```

Comment:
This might not be very close to the reality and that could make these tests quite brittle. Also, the this is failing for me, on my machine, running the tests from my IDE, min and max are both over 100ms.

I can see the test being more intuitive using the system clock but for robustness, I might use `MockClock`.